### PR TITLE
Deps bumps for 0.16.0

### LINKED ISSUE #2430: NextJS: 'Box' cannot be used as a JSX component.
**Describe the bug**
The problems appear on `next build` and seem to be related to all `theme-ui/Box` components:
```
./src/components/cards/Post.tsx:33:18
Type error: 'Heading' cannot be used as a JSX component.
  Its return type 'ReactNode' is not a valid JSX element.
```

**Screenshots**
This is what might cause the error:
https://github.com/system-ui/theme-ui/blob/6495a0a68f1a62bebc9f06fd650a11eaa9c2a3be/packages/components/src/Box.tsx#L134
image

```
'Component' cannot be used as a JSX component.
  Its element type 'ReactElement<any, any> | Component<any, any, any> | null' is not a valid JSX element.
    Type 'Component<any, any, any>' is not assignable to type 'Element | ElementClass | null'.
      Type 'Component<any, any, any>' is not assignable to type 'ElementClass'.
        The types returned by 'render()' are incompatible between these types.
          Type 'React.ReactNode' is not assignable to type 'import("/media/anton/Enterprise1/nesterov.digital/node_modules/@types/react/index").ReactNode'.
            Type 'ReactElement<any, string | JSXElementConstructor<any>>' is not assignable to type 'ReactNode'.
              Property 'children' is missing in type 'ReactElement<any, string | JSXElementConstructor<any>>' but required in type 'ReactPortal'.ts(2786)
```

**Additional context**
- OS: Ubuntu 22.10
- NodeJS v18.16.0
- `package.json`:
  ```
  "next": "13.4.4"
  "@types/node": "20.2.5"
  "@types/react": "18.2.8"
  "@types/react-dom": "18.2.4"
  ```
- `tsconfig.json`
  ```
  "jsxImportSource": "theme-ui",
  ```

### PR DESCRIPTION
**Pull request by hasparus**
I updated dependencies, including TypeScript to v5, and then I updated types for JSX namespace so it includes the new `ElementType`.

Fixes #2430, hopefully, but you need to remember to use automatic JSX runtime.

I'm thinking we should probably stop exporting `jsx` from the main packages to block the classic runtime, because it doesn't seem to jive well with the newest updates to React types and TS.

### Release notes

`@theme-ui/sidenav` package was removed due to low usage and breaking changes in `@types/react`.

# Version

Published prerelease version: `v0.16.0-develop.1`

<details>
  <summary>Changelog</summary>

  ### Release Notes
  
  #### Deps bumps for 0.16.0 ([#2432](https://github.com/system-ui/theme-ui/pull/2432))
  
  `@theme-ui/sidenav` package was removed due to low usage and breaking changes in `@types/react`.
  
  ---
  
  #### 🚀 Enhancement
  
  - Deps bumps for 0.16.0 [#2432](https://github.com/system-ui/theme-ui/pull/2432) ([@hasparus](https://github.com/hasparus))
  - Deprecate/rename ThemeProvider to ThemeUIProvider [#2360](https://github.com/system-ui/theme-ui/pull/2360) ([@lachlanjc](https://github.com/lachlanjc) [@hasparus](https://github.com/hasparus))
  
  #### 👨‍💻 Minor changes
  
  - Update lockfile ([@hasparus](https://github.com/hasparus))
  - Update Browserlist ([@hasparus](https://github.com/hasparus))
  - Update pnpm-lock.yaml ([@hasparus](https://github.com/hasparus))
  
  #### Authors: 2
  
  - Lachlan Campbell ([@lachlanjc](https://github.com/lachlanjc))
  - Piotr Monwid-Olechnowicz ([@hasparus](https://github.com/hasparus))
</details>

### CODE REVIEW

File: packages/docs/src/components/note.js [hunk 1]
```diff
@@ -1,6 +1,3 @@
-/** @jsx jsx */
-import { jsx } from 'theme-ui'
-
 export default (props) => (
```
Comment (lachlanjc) [inline packages/docs/src/components/note.js:1]:
```suggestion
// @ts-check

export default (props) => (
```

**Review by lachlanjc — APPROVED** [review]
Wow amazing, thank you for tackling! Everything looks good from reading the diff, haven't run anything locally though. We should probably add an announcement about `sidenav`'s demise to the changelog?

**Comment by hasparus** [comment]
I have some hydration warnings in the docs to fix, and the release notes for the removal of Sidenav, and I think we'll be good to go for 0.16

**Comment by hasparus** [comment]
:rocket: PR was released in [`v0.16.0`](https://github.com/system-ui/theme-ui/releases/tag/v0.16.0) :rocket:

**Pull request merged**